File: Lista-15/python/2.py
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self):
        self.velocidade += 10

    def frear(self):
        if self.velocidade >= 10:
            self.velocidade -= 10

    def exibir_velocidade(self):
        return f"Velocidade atual: {self.velocidade} km/h"

# Exemplo de uso
carro1 = Carro("Toyota", "Corolla", 2020)
carro1.acelerar()
print(carro1.exibir_velocidade())
carro1.frear()
print(carro1.exibir_velocidade())

File: Lista-15/go/2.go
package main
import "fmt"

type Carro struct {
    Marca      string
    Modelo     string
    Ano        int
    Velocidade int
}

func (c *Carro) Acelerar() {
    c.Velocidade += 10
}

func (c *Carro) Frear() {
    if c.Velocidade >= 10 {
        c.Velocidade -= 10
    }
}

func (c Carro) ExibirVelocidade() string {
    return fmt.Sprintf("Velocidade atual: %d km/h", c.Velocidade)
}

func main() {
    carro1 := Carro{"Toyota", "Corolla", 2020, 0}
    carro1.Acelerar()
    fmt.Println(carro1.ExibirVelocidade())
    carro1.Frear()
    fmt.Println(carro1.ExibirVelocidade())
}
